version-ranges/src/lib.rs

```diff
@@ -283,6 +290,37 @@ impl<V: Ord> Ranges<V> {
         }
     }
 
+    /// We want to use `iterator_try_collect`, but since it's unstable at the time of writing,
```

Comment:
This user facing documentation also needs to describe what this function does, and what the requirements are on the arguments, and when it returns an error.

Reply:
Yeah this comment seems more like an internal note?

Reply:
This isn't user facing, the function only exists internally for testing; if it weren't for the proptests this function wouldn't exist at all, `from_iter` is the real thing. I'll update the docs to redirect to `from_iter`.